tests/scripts/thread-cert/otci/otci/otci.py

```diff
@@ -0,0 +1,1315 @@
+
+    def get_eidcache(self) -> Dict[Ip6Addr, Rloc16]:
+        """Get the EID-to-RLOC cache entries."""
+        output = self.execute_command('eidcache')
+        cache = {}
+
+        for line in output:
+            ip, rloc16 = line.split()
+
+            cache[Ip6Addr(ip)] = Rloc16(rloc16, 16)
+
+        return cache
+
+    #
+    # Other TODOs
+    #
+    # TODO: netstat
+    # TODO: networkdiagnostic get <addr> <type> ..
+    # TODO: networkdiagnostic reset <addr> <type> ..
+    # TODO: parent
+    # TODO: pskc [-p] <key>|<passphrase>
+    #
+
+    #
+    # Private methods
+    #
+
+    def __parse_str(self, output: List[str]) -> str:
+        if len(output) != 1:
+            raise UnexpectedCommandOutput(output)
+
+        return output[0]
+
+    def __parse_ip6addr(self, output: List[str]) -> Ip6Addr:
+        return Ip6Addr(self.__parse_str(output))
+
+    def __parse_int(self, output: List[str], base=10) -> int:
+        if len(output) != 1:
+            raise UnexpectedCommandOutput(output)
+
+        return int(output[0], base)
+
+    def __parse_master_key(self, output: List[str]) -> str:
+        masterkey = self.__parse_str(output)
+
+        try:
+            self.__validate_master_key(masterkey)
+        except ValueError:
+            raise UnexpectedCommandOutput(output)
+
+        return masterkey
+
+    def __validate_master_key(self, masterkey: str):
+        if len(masterkey) != 32:
+            raise ValueError(masterkey)
+
+        int(masterkey, 16)
+
+    def __parse_hex64b(self, output: List[str]) -> str:
+        extaddr = self.__parse_str(output)
+
+        try:
+            self.__validate_hex64b(extaddr)
+        except ValueError:
+            raise UnexpectedCommandOutput(output)
+
+        return extaddr
+
+    __parse_extaddr = __parse_hex64b
+    __parse_extpanid = __parse_hex64b
+    __parse_eui64 = __parse_hex64b
+    __parse_joiner_id = __parse_hex64b
+
+    def __validate_hex64b(self, extaddr: str):
+        if len(extaddr) != 16:
+            raise ValueError(extaddr)
+
+        int(extaddr, 16)
+
+    __validate_extaddr = __validate_hex64b
+    __validate_extpanid = __validate_hex64b
+
+    def __parse_values(self, output: List[str], **vals) -> Any:
+        val = self.__parse_str(output)
+        if val not in vals:
+            raise UnexpectedCommandOutput(output)
+
+        return vals[val]
+
+    def __escape_escapable(self, s: str) -> str:
+        """Escape CLI escapable characters in the given string.
+        """
+        escapable_chars = '\\ \t\r\n'
+        for char in escapable_chars:
+            s = s.replace(char, '\\%s' % char)
+        return s
+
+    def __wait_responsive(self):
+        timeout = 10
+        for i in range(timeout):
+            try:
+                self.execute_command('version', 1, silent=True)
+                break
+            except OTCIError:
+                if i == timeout - 1:
+                    raise
+
```

Comment:
Seems we need to add a delay before next try.

Reply:
Yes. I think we need to delay before each try (including the first try).